pkg/controller/operands/cdi_test.go

```diff
@@ -382,6 +384,137 @@ var _ = Describe("CDI Operand", func() {
 			Expect(foundResource.Spec.Config.FeatureGates).To(ContainElement("HonorWaitForFirstConsumer"))
 		})
 
+		It("should add cert configuration if missing in CDI", func() {
+			expectedResource, err := NewCDI(hco)
+			Expect(err).ToNot(HaveOccurred())
+			expectedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/%s/dummies/%s", expectedResource.Namespace, expectedResource.Name)
+
+			hco.Spec.CertConfig = &hcov1beta1.HyperConvergedCertConfig{
+				CA: &hcov1beta1.CertRotateConfig{
+					Duration:    metav1.Duration{Duration: 1 * time.Hour},
+					RenewBefore: metav1.Duration{Duration: 2 * time.Hour},
+				},
+				Server: &hcov1beta1.CertRotateConfig{
+					Duration:    metav1.Duration{Duration: 3 * time.Hour},
+					RenewBefore: metav1.Duration{Duration: 4 * time.Hour},
+				},
+			}
+
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, expectedResource})
+			handler := (*genericOperand)(newCdiHandler(cl, commonTestUtils.GetScheme()))
+			res := handler.ensure(req)
+
+			Expect(res.UpgradeDone).To(BeFalse())
+			Expect(res.Updated).To(BeTrue())
+			Expect(res.Err).To(BeNil())
+
+			foundResource := &cdiv1beta1.CDI{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Name: expectedResource.Name, Namespace: expectedResource.Namespace},
+					foundResource),
+			).To(BeNil())
+
+			Expect(expectedResource.Spec.CertConfig).To(BeNil())
+			Expect(foundResource.Spec.CertConfig).ToNot(BeNil())
+			Expect(foundResource.Spec.CertConfig.CA.Duration.Duration.String()).Should(Equal("1h0m0s"))
+			Expect(foundResource.Spec.CertConfig.CA.RenewBefore.Duration.String()).Should(Equal("2h0m0s"))
+			Expect(foundResource.Spec.CertConfig.Server.Duration.Duration.String()).Should(Equal("3h0m0s"))
+			Expect(foundResource.Spec.CertConfig.Server.RenewBefore.Duration.String()).Should(Equal("4h0m0s"))
+			Expect(req.Conditions).To(BeEmpty())
+		})
+
+		It("should remove cert configuration if missing in HCO CR", func() {
```

Comment:
I think we should revisit this to be coherent with the default use case

Reply:
meaning not removing the configuration, but rather reverting to using the defaults?

Reply:
Yes, exactly.
A missing configuration on HCO CR should set HCO opinionated and unified defaults on all the components.

Reply:
done